synapse/lib/drive.py

```diff
@@ -60,24 +61,32 @@ def getPathNorm(self, path):
 
         return [reqValidName(p.strip().lower()) for p in path]
 
-    def getItemInfo(self, iden):
-        return self._getItemInfo(s_common.uhex(iden))
+    def getItemInfo(self, iden, typename=None):
+        info = self._getItemInfo(s_common.uhex(iden))
+        if typename is not None and info.get('type') != typename:
+            infotype = info.get('type')
+            mesg = f'Drive item has the wrong type. Expected: {typename} got {infotype}.'
+            raise s_exc.TypeMismatch(mesg=mesg)
+        return info
 
     def _getItemInfo(self, bidn):
         byts = self.slab.get(LKEY_INFO + bidn, db=self.dbname)
         if byts is not None:
             return s_msgpack.un(byts)
 
-    def reqItemInfo(self, iden):
-        return self._reqItemInfo(s_common.uhex(iden))
+    def reqItemInfo(self, iden, typename=None):
+        return self._reqItemInfo(s_common.uhex(iden), typename=typename)
 
-    def _reqItemInfo(self, bidn):
+    def _reqItemInfo(self, bidn, typename=None):
         info = self._getItemInfo(bidn)
-        if info is not None:
-            return info
+        if info is None:
+            mesg = f'No drive item with ID {s_common.ehex(bidn)}.'
+            raise s_exc.NoSuchIden(mesg=mesg)
+
+        if typename is not None and info.get('type') != typename:
+            raise s_exc.SynErr(mesg='Drive item has the wrong type.')
```

Comment:
Typemismatch  ?

Reply:
thanks, the TypeMismatch exc was a last minute add :+1: